cql3/functions/user_function.cc

```diff
@@ -73,31 +75,23 @@ description user_function::describe(with_create_statement with_stmt) const {
             return std::nullopt;
         }
 
-        auto ks = cql3::util::maybe_quote(name().keyspace);
-        auto na = cql3::util::maybe_quote(name().name);
-
-        std::ostringstream os;
-
-        os << "CREATE FUNCTION " << ks << "." << na << "(";
-        for (size_t i = 0; i < _arg_names.size(); i++) {
-            if (i > 0) {
-                os << ", ";
-            }
-            os << _arg_names[i] << " " << _arg_types[i]->cql3_type_name();
-        }
-        os << ")\n";
-
-        if (_called_on_null_input) {
-            os << "CALLED";
-        } else {
-            os << "RETURNS NULL";
-        }
-        os << " ON NULL INPUT\n"
-           << "RETURNS " << _return_type->cql3_type_name() << "\n"
-           << "LANGUAGE " << _language << "\n"
-           << "AS $$" << _body << "$$;";
-
-        return std::move(os).str();
+        auto arg_type_range = _arg_types | std::views::transform(std::mem_fn(&abstract_type::cql3_type_name));
+        auto arg_range = std::views::zip(_arg_names, arg_type_range)
```

Comment:
I have to ask this: Do you believe this code, a long string of functions calls like transform (two of them), mem_fn, zip, and then a manual "unzip" - all of this is really "easier to read", than a 1970s C for loop and C++'s 1990s  output operators?

So I'm curious what motivated this patch. I'm not strongly against it, but it seems unnecessary for the following patch - it just replaces cql3_type_name() by cql3_type_name_without_frozen(), which was equally easy to do before this refactoring and after it.

Reply:
Ok. I don't have any defensible objection to this change.